src/Uno.UI/UI/Xaml/Controls/StackPanel/StackPanel.Layout.cs

```diff
@@ -134,7 +134,7 @@ protected override Size ArrangeOverride(Size arrangeSize)
 					childRectangle.Width = desiredChildSize.Width;
 					childRectangle.Height = Math.Max(arrangeSize.Height, desiredChildSize.Height);
 
-					var snapPoint = (float)childRectangle.Right;
+					var snapPoint = (float)(childRectangle.Right + Margin.Right);
```

Comment:
If we don't have any tests validating this already, let's add some.

Reply:
The test I added in `Given_StackPanel` should be validating this.